### DISCUSSION TITLE
CORS policy: No 'Access-Control-Allow-Origin' header

### DISCUSSION THREAD

**Discussion by jktan0504** [devlikeapro/waha#405] [General]
channel:1 Access to fetch at 'my-waha' from origin 'my-frontend' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.

`server {
listen 80;
listen [::]:80;

          listen 443 ssl;
          listen [::]:443 ssl;

          ssl_certificate /etc/nginx/certs/cert.pem;
          ssl_certificate_key /etc/nginx/certs/private.pem;

          server_name mydomain.com;

          location / {
                proxy_pass http://waha_plus:3000;  # Name of the API service defined in docker-compose.yml
                proxy_set_header Host $host;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Proto $scheme;

				# CORS configuration
				add_header 'Access-Control-Allow-Origin' 'https://myfrontend.com' always;
				add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS, DELETE, PUT, PATCH' always;
				add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization' always;
				add_header 'Access-Control-Allow-Credentials' 'true' always;

				if ($request_method = 'OPTIONS') {
					add_header 'Access-Control-Allow-Origin' 'https://myfrontend.com' always;
					add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS, DELETE, PUT, PATCH' always;
					add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization' always;
					add_header 'Access-Control-Max-Age' 1728000;
					add_header 'Content-Type' 'text/plain; charset=utf-8';
					add_header 'Content-Length' 0;
					return 204;
				}
          }
    }`

Anyone ?

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
WAHA allows CORS by default, you can test it by accessing your own instance from different domain like in a browser. Kindly check your reverse proxy configuraiton (I'd suggest using something like `dokku` or Portainer with Traefik, it handles such cases)
https://waha.devlike.pro/dashboard/

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
